### PR TITLE
Dmackenzie/us105618 focus on quick eval table

### PR DESCRIPTION
**Pull request by dmackenz**

### CODE REVIEW

File: components/d2l-quick-eval/d2l-quick-eval.js [hunk 1]
```diff
@@ -126,6 +128,10 @@ class D2LQuickEval extends
 			},
 			token: {
 				type: Object,
+			},
+			returningToQuickEval: {
```
Comment (dmackenz) [inline components/d2l-quick-eval/d2l-quick-eval.js]: Gets set from `EvaluationHubController` now, no longer scraped from `window.location`.

**Comment by dmackenz** [comment]
Linked with https://git.dev.d2l/projects/CORE/repos/lms/pull-requests/6666/overview

File: components/d2l-quick-eval/d2l-quick-eval-submissions-table.js [hunk 2]
```diff
@@ -320,6 +326,19 @@ class D2LQuickEvalSubmissionsTable extends QuickEvalLogging(QuickEvalLocalize(Po
 		}
 	}
 
+	_handleNameFocusOnPageForwardBack() {
+		if (((window.performance && window.performance.navigation.type === window.performance.navigation.TYPE_BACK_FORWARD) || (this.returningToQuickEval === true)) && !this.showLoadingSkeleton && this._data.length > 0) {
+			this.async(() => {
```
Comment (dmackenz) [inline components/d2l-quick-eval/d2l-quick-eval-submissions-table.js]: These asyncs look weird, but they are related to the fact that the element that I am targetting is in the dom-repeat and is dynamically rendered. Got this solution from here:

https://stackoverflow.com/questions/44141972/polymer-queryselector-cant-find-element-in-dom-repeat

https://stackoverflow.com/questions/30081215/polymer-when-to-use-async

File: components/d2l-quick-eval/d2l-quick-eval-submissions-table.js [hunk 3]
```diff
@@ -320,6 +326,19 @@ class D2LQuickEvalSubmissionsTable extends QuickEvalLogging(QuickEvalLocalize(Po
 		}
 	}
 
+	_handleNameFocusOnPageForwardBack() {
+		if (((window.performance && window.performance.navigation.type === window.performance.navigation.TYPE_BACK_FORWARD) || (this.returningToQuickEval === true)) && !this.showLoadingSkeleton && this._data.length > 0) {
```
Comment (jeroach) [inline components/d2l-quick-eval/d2l-quick-eval-submissions-table.js]: Hmmm, maybe try this on multiple lines? or split up into assigned variables with intermediate condition names?

seems really hard to understand the logic as it stands

Comment (dmackenz) [reply components/d2l-quick-eval/d2l-quick-eval-submissions-table.js]: Good point with the variables! will do.

Comment (jeroach) [reply components/d2l-quick-eval/d2l-quick-eval-submissions-table.js]: Awesome, thanks :)

File: test/d2l-quick-eval/d2l-quick-eval-submissions-table.js [hunk 4]
```diff
@@ -367,6 +367,16 @@ import '@polymer/iron-test-helpers/mock-interactions.js';
 
 			});
 		});
+		test('Ensure that focus is on the first name of the submissions table when returning to quick eval', (done) => {
+			list.returningToQuickEval = true
```
Comment (jeroach) [inline test/d2l-quick-eval/d2l-quick-eval-submissions-table.js]: Missing some semicolons in here:
https://travis-ci.com/BrightspaceHypermediaComponents/activities/jobs/268718615#L244

**Pull request merged**